98 함께 전시 관람 신청 목록 조회

chlalswns200 opened the pull request:
## :information_desk_person: 간단 소개
enroll/list: 승인 받지 않은 participant 목록을 조회 합니다(글 작성자만 조회 가능)
approve/list: 승인 받은 participant 목록을 조회 합니다

## :heavy_check_mark: 작업 내용 설명
### Controller
  (GET) enroll/list 메서드 추가
  (GET) approve/list 메서드 추가
### Service
  enrollList() 메서드 추가
  approveList() 메서드 추가
### Repository
  findAllByGatheringIdAndApprove() 메서드 추가
### Entity
  toEnrollResponse() 메서드 추가
### Dto
  GatheringResponse 추가

### Test
  성공
  실패-1 이메일 조회 실패  
  실패-2 모집 글 조회 실패

## :bulb: 참고사항
로직 변경 등 리뷰어나 다른 팀원들이 참고해야 할 사항을 적어주세요.

chlalswns200 committed:
feat 함께 전시 관람 목록 조회 기능 추가
Controller
  (GET) enroll/list 메서드 추가
  (GET) approve/list 메서드 추가
Service
  enrollList() 메서드 추가
  approveList() 메서드 추가
Repository
  findAllByGatheringIdAndApprove() 메서드 추가
Entity
  toEnrollResponse() 메서드 추가

chlalswns200 committed:
feat 함께 전시 관람 목록 조회 기능 추가
Dto
  GatheringResponse 추가

chlalswns200 committed:
test: 참가 신청 목록 조회 테스트
성공
실패#1 이메일 조회 실패
실패#2 모집 글 조회 실패

chlalswns200 committed:
Merge branch 'develop' into 98-함께_전시_관람_신청_목록_조회

geuun commented on src/main/java/com/dev/museummate/domain/entity/ParticipantEntity.java:
```diff
@@ -44,4 +45,15 @@ public ParticipantEntity(UserEntity user, GatheringEntity gathering, Boolean hos
         this.approve = approve;
     }
+
+    public GatheringResponse toEnrollResponse() {
```
이 부분 Response로 바로 매핑 하게되면 안될 것 같습니다!

Dto로 매핑되게 해주셔야 할 것 같아요

geuun commented on src/main/java/com/dev/museummate/service/GatheringService.java:
```diff
@@ -59,4 +62,35 @@ public String enroll(Long gatheringId, String email) {
         return "신청이 완료 되었습니다.";
     }
+
+    public List<GatheringResponse> enrollList(Long gatheringId, String email) {
```
Service 로직도 다른 도메인들과 일관성있게 비즈니스로직에서 Dto로 반환 하고
Controller에서 Response에 매핑하게 리펙토링 부탁드립니다!

ahn-ji0 reviewed (approved):
확인했습니다!

geuun commented:
@chlalswns200 

민준님 리뷰 코멘트 확인 부탁드릴게요!

chlalswns200 committed:
refacto: service 반환 dto로 변경
Controller
  service에서 dto 받아서 response 만드는 방식으로 수정
GatheringEntity
  name에 대문자 변경
Service
  entity->response에서 entity->dto로 변경 후 controller에 반환
ParticipantEntity
  toEnrollResponse()-> toDto()로 변경
ParticipantDto
  클래스 생성

chlalswns200 committed:
test: service 반환 dto로 변경
enrollList_success:
  List<GatheringResponse> -> List<ParticipantDto>

chlalswns200 commented:
말씀해주신 내용 반영했습니다
다시 확인 부탁드립니다!

geuun reviewed (approved):
리펙토링 확인했습니다!

수고하셨습니다 ~ !

geuun commented:
병합 진행하겠습니다  😎 👍
